src/Compilers/CSharp/Test/Emit3/PartialEventsAndConstructorsTests.cs

```diff
@@ -946,6 +946,313 @@ partial interface I
+            Assert.True(m.ContainingModule is not SourceModuleSymbol || m.IsPartialMember());
+        }
+    }
+
+    [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/77346")]
+    public void InInterface_Sealed(
+        [CombinatorialValues("", "public")] string access)
+    {
+        var source = $$"""
+            partial interface I
+            {
+                {{access}} sealed partial event System.Action E;
+                {{access}} sealed partial event System.Action E { add { } remove { } }
+            }
+            """;
+        var comp = CreateCompilation(source, targetFramework: TargetFramework.Net60).VerifyDiagnostics();
+        CompileAndVerify(comp,
+            sourceSymbolValidator: validate,
+            symbolValidator: validate,
+            verify: Verification.FailsPEVerify).VerifyDiagnostics();
+
+        static void validate(ModuleSymbol module)
+        {
+            var e = module.GlobalNamespace.GetMember<EventSymbol>("I.E");
+            validateEvent(e);
+
+            if (module is SourceModuleSymbol)
+            {
+                validateEvent((EventSymbol)e.GetPartialImplementationPart()!);
+            }
+        }
+
+        static void validateEvent(EventSymbol e)
+        {
+            Assert.False(e.IsAbstract);
+            Assert.False(e.IsVirtual);
+            Assert.False(e.IsSealed);
+            Assert.False(e.IsStatic);
+            Assert.False(e.IsExtern);
+            Assert.False(e.IsOverride);
+            Assert.Equal(Accessibility.Public, e.DeclaredAccessibility);
+            Assert.True(e.ContainingModule is not SourceModuleSymbol || e.IsPartialMember());
+            validateAccessor(e.AddMethod!);
+            validateAccessor(e.RemoveMethod!);
+        }
+
+        static void validateAccessor(MethodSymbol m)
+        {
+            Assert.False(m.IsAbstract);
+            Assert.False(m.IsVirtual);
+            Assert.False(m.IsMetadataVirtual());
+            Assert.False(m.IsMetadataNewSlot());
+            Assert.False(m.IsSealed);
+            Assert.False(m.IsStatic);
+            Assert.False(m.IsExtern);
+            Assert.False(m.IsOverride);
+            Assert.Equal(Accessibility.Public, m.DeclaredAccessibility);
+            Assert.True(m.ContainingModule is not SourceModuleSymbol || m.IsPartialMember());
+        }
+    }
+
+    [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/77346")]
+    public void InInterface_StaticVirtual(
+        [CombinatorialValues("", "public")] string access)
+    {
+        var source = $$"""
+            using System;
+
+            M<C1>();
+            M<C2>();
+            M<C3>();
+
+            static void M<T>() where T : I
+            {
+                T.E += null;
+                T.E -= null;
+            }
+
+            partial interface I
+            {
+                {{access}} static virtual partial event Action E;
```

Comment:
>virtual [](http://example.com/codeflow?start=34&length=7)

It would be good to test:
- without `virtual`, should remain non-virtual
- without `virtual` and with `sealed`

Reply:
For other member kinds too